src/lib/tls/tls_server.cpp

```diff
@@ -1020,6 +1064,126 @@ void Server::session_create(Server_Handshake_State& pending_state,
 
    pending_state.server_hello_done(new Server_Hello_Done(pending_state.handshake_io(), pending_state.hash()));
    }
+
+DTLS_Prestate Server::pre_verify_cookie(Credentials_Manager& creds,
+                               const Policy& policy,
+                               const std::string& client_identity,
+                               const uint8_t input[],
+                               size_t buf_size,
+                               output_fn emit_data_fn)
+   {
+   secure_vector<uint8_t> readbuf;
+   secure_vector<uint8_t> record_buf;
+   Stream_Sequence_Numbers seq;
+   // a hacky use of Stream_Sqeuence_Numbers
+   SymmetricKey cookie_secret;
+   size_t consumed = 0;
+
+   const Record_Header record =
+      read_record(true, readbuf, input,
+                  buf_size, consumed,
+                  record_buf, &seq, [](uint16_t) -> std::shared_ptr<Connection_Cipher_State> {return nullptr;},
```

Comment:
Eek yeah I don't like this use of dummy seq, a refactor of the record reader code would help and also avoid the no-op lambda.

Reply:
Actually I think you can just use `nullptr` for the seq here